File: LCX/ValidEndPoints/dsl/examples/trading_operations/trading_operations.nim
import httpclient
import json
import hmac
import base64


# POST method not supported for nim
# POST method not supported for nim
# POST method not supported for nim
# PUT method not supported for nim
# PUT method not supported for nim
# PUT method not supported for nim
# DELETE method not supported for nim
# DELETE method not supported for nim
proc get_all_open_orders*(offset): JsonNode =
  ## GET /api/open
  let url = "https://exchange-api.lcx.com/api/open"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_open_orders_for_pair*(offset, pair): JsonNode =
  ## GET /api/open
  let url = "https://exchange-api.lcx.com/api/open"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_open_orders_with_date_range*(offset, pair, fromDate, toDate): JsonNode =
  ## GET /api/open
  let url = "https://exchange-api.lcx.com/api/open"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_single_order_status*(orderId): JsonNode =
  ## GET /api/order
  let url = "https://exchange-api.lcx.com/api/order"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_complete_order_history*(offset): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_order_history_for_pair*(offset, pair): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_order_history_by_status*(offset, orderStatus): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_order_history_by_type*(offset, orderType): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_filtered_order_history*(offset, pair, fromDate, toDate, side, orderStatus, orderType): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_all_trade_history*(offset): JsonNode =
  ## GET /api/uHistory
  let url = "https://exchange-api.lcx.com/api/uHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_trade_history_for_pair*(offset, pair): JsonNode =
  ## GET /api/uHistory
  let url = "https://exchange-api.lcx.com/api/uHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_trade_history_by_date_range*(offset, pair, fromDate, toDate): JsonNode =
  ## GET /api/uHistory
  let url = "https://exchange-api.lcx.com/api/uHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc simple_buy_workflow*(order_result): JsonNode =
  ## GET /api/order
  let url = "https://exchange-api.lcx.com/api/order"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# Workflow: SimpleBuyWorkflow (sequence)

# None method not supported for nim
# PUT method not supported for nim
proc get_history*(0, BTC/USDC): JsonNode =
  ## GET /api/orderHistory
  let url = "https://exchange-api.lcx.com/api/orderHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

proc get_trades*(0, BTC/USDC): JsonNode =
  ## GET /api/uHistory
  let url = "https://exchange-api.lcx.com/api/uHistory"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# None method not supported for nim
# DELETE method not supported for nim
proc confirm_cancelled*(first_order_from_list): JsonNode =
  ## GET /api/order
  let url = "https://exchange-api.lcx.com/api/order"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)

  return data

# None method not supported for nim

File: LCX/ValidEndPoints/dsl/examples/trading_operations/trading_operations.lua
local http = require("socket.http")
local json = require("cjson")


# POST method not supported for lua
# POST method not supported for lua
# POST method not supported for lua
# PUT method not supported for lua
# PUT method not supported for lua
# PUT method not supported for lua
# DELETE method not supported for lua
# DELETE method not supported for lua
function get_all_open_orders(offset)
  -- GET /api/open
  local url = "https://exchange-api.lcx.com/api/open"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_open_orders_for_pair(offset, pair)
  -- GET /api/open
  local url = "https://exchange-api.lcx.com/api/open"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_open_orders_with_date_range(offset, pair, fromDate, toDate)
  -- GET /api/open
  local url = "https://exchange-api.lcx.com/api/open"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_single_order_status(orderId)
  -- GET /api/order
  local url = "https://exchange-api.lcx.com/api/order"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_complete_order_history(offset)
  -- GET /api/orderHistory
  local url = "https://exchange-api.lcx.com/api/orderHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_order_history_for_pair(offset, pair)
  -- GET /api/orderHistory
  local url = "https://exchange-api.lcx.com/api/orderHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_order_history_by_status(offset, orderStatus)
  -- GET /api/orderHistory
  local url = "https://exchange-api.lcx.com/api/orderHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_order_history_by_type(offset, orderType)
  -- GET /api/orderHistory
  local url = "https://exchange-api.lcx.com/api/orderHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_filtered_order_history(offset, pair, fromDate, toDate, side, orderStatus, orderType)
  -- GET /api/orderHistory
  local url = "https://exchange-api.lcx.com/api/orderHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_all_trade_history(offset)
  -- GET /api/uHistory
  local url = "https://exchange-api.lcx.com/api/uHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_trade_history_for_pair(offset, pair)
  -- GET /api/uHistory
  local url = "https://exchange-api.lcx.com/api/uHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_trade_history_by_date_range(offset, pair, fromDate, toDate)
  -- GET /api/uHistory
  local url = "https://exchange-api.lcx.com/api/uHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function simple_buy_workflow(order_result)
  -- GET /api/order
  local url = "https://exchange-api.lcx.com/api/order"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

# Workflow: SimpleBuyWorkflow (sequence)

# None method not supported for lua
# PUT method not supported for lua
function get_history(0, BTC/USDC)
  -- GET /api/orderHistory
  local url = "https://exchange-api.lcx.com/api/orderHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

function get_trades(0, BTC/USDC)
  -- GET /api/uHistory
  local url = "https://exchange-api.lcx.com/api/uHistory"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

# None method not supported for lua
# DELETE method not supported for lua
function confirm_cancelled(first_order_from_list)
  -- GET /api/order
  local url = "https://exchange-api.lcx.com/api/order"
  local http = require("socket.http")

  local body, code = http.request(url)
  if code == 200 then

  else
    error("HTTP " .. code)
  end
end

# None method not supported for lua
